// @ngx-gallery/core/esm2015/components/gallery-counter.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
export class GalleryCounterComponent {
}
GalleryCounterComponent.decorators = [
    { type: Component, args: [{
                selector: 'gallery-counter',
                changeDetection: ChangeDetectionStrategy.OnPush,
                preserveWhitespaces: false,
                template: `<div class="g-counter">{{(state.currIndex + 1) + '/' + state.items.length}}</div>`
            },] },
];
/** @nocollapse */
GalleryCounterComponent.ctorParameters = () => [];
GalleryCounterComponent.propDecorators = {
    "state": [{ type: Input },],
};
function GalleryCounterComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    GalleryCounterComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    GalleryCounterComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    GalleryCounterComponent.propDecorators;
    /** @type {?} */
    GalleryCounterComponent.prototype.state;
}
