// @ngx-gallery/core/esm5/templates/iframe-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
var IframeItemComponent = /** @class */ (function () {
    function IframeItemComponent(sanitizer) {
        this.sanitizer = sanitizer;
    }
    IframeItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'iframe-item',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    preserveWhitespaces: false,
                    template: "\n    <iframe frameborder=\"0\"\n            allowfullscreen\n            [src]=\"sanitizer.bypassSecurityTrustResourceUrl(data?.src)\"></iframe>\n  ",
                    styles: ["\n    iframe {\n      position: absolute;\n      width: 100%;\n      height: 100%;\n    }\n  "]
                },] },
    ];
    /** @nocollapse */
    IframeItemComponent.ctorParameters = function () { return [
        { type: DomSanitizer, },
    ]; };
    IframeItemComponent.propDecorators = {
        "data": [{ type: Input },],
    };
    return IframeItemComponent;
}());
export { IframeItemComponent };
function IframeItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    IframeItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    IframeItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    IframeItemComponent.propDecorators;
    /** @type {?} */
    IframeItemComponent.prototype.data;
    /** @type {?} */
    IframeItemComponent.prototype.sanitizer;
}
