// @ngx-gallery/core/esm5/templates/image-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { Gallery } from '../services/gallery.service';
var ImageItemComponent = /** @class */ (function () {
    function ImageItemComponent(gallery, sanitizer) {
        this.gallery = gallery;
        this.sanitizer = sanitizer;
    }
    ImageItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'image-item',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    preserveWhitespaces: false,
                    template: "\n    <div [lazyImage]=\"data?.src\" (loading)=\"loading = $event\"></div>\n    <i class=\"g-loading\" *ngIf=\"loading\" [innerHTML]=\"sanitizer.bypassSecurityTrustHtml(gallery.config.loadingIcon)\"></i>\n  ",
                    styles: ["\n    :host {\n      position: relative;\n      display: block;\n      width: 100%;\n      height: 100%;\n    }\n    div {\n      background-position: center center;\n      background-size: cover;\n      width: 100%;\n      height: 100%;\n    }\n  "],
                },] },
    ];
    /** @nocollapse */
    ImageItemComponent.ctorParameters = function () { return [
        { type: Gallery, },
        { type: DomSanitizer, },
    ]; };
    ImageItemComponent.propDecorators = {
        "data": [{ type: Input },],
    };
    return ImageItemComponent;
}());
export { ImageItemComponent };
function ImageItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    ImageItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    ImageItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    ImageItemComponent.propDecorators;
    /** @type {?} */
    ImageItemComponent.prototype.loading;
    /** @type {?} */
    ImageItemComponent.prototype.data;
    /** @type {?} */
    ImageItemComponent.prototype.gallery;
    /** @type {?} */
    ImageItemComponent.prototype.sanitizer;
}
