// @ngx-gallery/core/esm5/templates/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ImageItemComponent } from './image-item.component';
import { VideoItemComponent } from './video-item.component';
import { IframeItemComponent } from './iframe-item.component';
import { ThumbnailItemComponent } from './thumbnail-item.component';
var ImageItem = /** @class */ (function () {
    function ImageItem(src, thumb) {
        this.component = ImageItemComponent;
        this.thumbComponent = ThumbnailItemComponent;
        this.data = { src: src, thumb: thumb };
    }
    return ImageItem;
}());
export { ImageItem };
function ImageItem_tsickle_Closure_declarations() {
    /** @type {?} */
    ImageItem.prototype.component;
    /** @type {?} */
    ImageItem.prototype.thumbComponent;
    /** @type {?} */
    ImageItem.prototype.data;
}
var VideoItem = /** @class */ (function () {
    function VideoItem(src, thumb, poster) {
        this.component = VideoItemComponent;
        this.thumbComponent = ThumbnailItemComponent;
        this.data = { src: src, thumb: thumb, poster: poster };
    }
    return VideoItem;
}());
export { VideoItem };
function VideoItem_tsickle_Closure_declarations() {
    /** @type {?} */
    VideoItem.prototype.component;
    /** @type {?} */
    VideoItem.prototype.thumbComponent;
    /** @type {?} */
    VideoItem.prototype.data;
}
var IframeItem = /** @class */ (function () {
    function IframeItem(src, thumb) {
        this.component = IframeItemComponent;
        this.thumbComponent = ThumbnailItemComponent;
        this.data = { src: src, thumb: thumb };
    }
    return IframeItem;
}());
export { IframeItem };
function IframeItem_tsickle_Closure_declarations() {
    /** @type {?} */
    IframeItem.prototype.component;
    /** @type {?} */
    IframeItem.prototype.thumbComponent;
    /** @type {?} */
    IframeItem.prototype.data;
}
var YoutubeItem = /** @class */ (function () {
    function YoutubeItem(src, thumb) {
        this.component = IframeItemComponent;
        this.thumbComponent = ThumbnailItemComponent;
        this.data = {
            src: "//youtube.com/embed/" + src + "?wmode=transparent",
            thumb: thumb ? thumb : "//img.youtube.com/vi/" + src + "/default.jpg"
        };
    }
    return YoutubeItem;
}());
export { YoutubeItem };
function YoutubeItem_tsickle_Closure_declarations() {
    /** @type {?} */
    YoutubeItem.prototype.component;
    /** @type {?} */
    YoutubeItem.prototype.thumbComponent;
    /** @type {?} */
    YoutubeItem.prototype.data;
}
export { ImageItemComponent, VideoItemComponent, IframeItemComponent, ThumbnailItemComponent };
