// @ngx-gallery/core/esm5/templates/thumbnail-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
var ThumbnailItemComponent = /** @class */ (function () {
    function ThumbnailItemComponent() {
    }
    ThumbnailItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'thumbnail-item',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    preserveWhitespaces: false,
                    template: "\n    <div [lazyImage]=\"data.thumb\"></div>\n  ",
                    styles: ["\n    :host {\n      width: 100%;\n      height: 100%;\n    }\n    div {\n      width: 100%;\n      height: 100%;\n      background-size: cover;\n      background-position: center center;\n    }\n  "]
                },] },
    ];
    /** @nocollapse */
    ThumbnailItemComponent.ctorParameters = function () { return []; };
    ThumbnailItemComponent.propDecorators = {
        "data": [{ type: Input },],
    };
    return ThumbnailItemComponent;
}());
export { ThumbnailItemComponent };
function ThumbnailItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    ThumbnailItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    ThumbnailItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    ThumbnailItemComponent.propDecorators;
    /** @type {?} */
    ThumbnailItemComponent.prototype.data;
}
