// @ngx-gallery/core/esm5/templates/video-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
var VideoItemComponent = /** @class */ (function () {
    function VideoItemComponent(sanitizer) {
        this.sanitizer = sanitizer;
    }
    VideoItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'video-item',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    preserveWhitespaces: false,
                    template: "\n    <video controls\n           [src]=\"sanitizer.bypassSecurityTrustResourceUrl(data?.src)\"\n           [poster]=\"sanitizer.bypassSecurityTrustResourceUrl(data?.poster)\"></video>\n  ",
                    styles: ["\n    video {\n      position: absolute;\n      width: 100%;\n      height: 100%;\n    }\n  "]
                },] },
    ];
    /** @nocollapse */
    VideoItemComponent.ctorParameters = function () { return [
        { type: DomSanitizer, },
    ]; };
    VideoItemComponent.propDecorators = {
        "data": [{ type: Input },],
    };
    return VideoItemComponent;
}());
export { VideoItemComponent };
function VideoItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    VideoItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    VideoItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    VideoItemComponent.propDecorators;
    /** @type {?} */
    VideoItemComponent.prototype.data;
    /** @type {?} */
    VideoItemComponent.prototype.sanitizer;
}
