// @ngx-gallery/core/esm5/components/gallery-nav.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
var GalleryNavComponent = /** @class */ (function () {
    function GalleryNavComponent() {
        this.indexChange = new EventEmitter();
    }
    GalleryNavComponent.decorators = [
        { type: Component, args: [{
                    selector: 'gallery-nav',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    preserveWhitespaces: false,
                    template: "\n    <i *ngIf=\"config.loop || state.hasPrev\" class=\"g-nav-prev\" (tapClick)=\"indexChange.emit('prev')\"></i>\n    <i *ngIf=\"config.loop || state.hasNext\" class=\"g-nav-next\" (tapClick)=\"indexChange.emit('next')\"></i>\n  "
                },] },
    ];
    /** @nocollapse */
    GalleryNavComponent.ctorParameters = function () { return []; };
    GalleryNavComponent.propDecorators = {
        "state": [{ type: Input },],
        "config": [{ type: Input },],
        "indexChange": [{ type: Output },],
    };
    return GalleryNavComponent;
}());
export { GalleryNavComponent };
function GalleryNavComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    GalleryNavComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    GalleryNavComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    GalleryNavComponent.propDecorators;
    /** @type {?} */
    GalleryNavComponent.prototype.state;
    /** @type {?} */
    GalleryNavComponent.prototype.config;
    /** @type {?} */
    GalleryNavComponent.prototype.indexChange;
}
