// @ngx-gallery/core/esm5/directives/gallery-item.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, Input, ComponentFactoryResolver, ViewContainerRef } from '@angular/core';
var GalleryItemDirective = /** @class */ (function () {
    function GalleryItemDirective(componentFactoryResolver, viewContainerRef) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.viewContainerRef = viewContainerRef;
        this.type = 'slide';
    }
    /**
     * @return {?}
     */
    GalleryItemDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ componentFactory = this.componentFactoryResolver
            .resolveComponentFactory(this.item[this.type === 'slide' ? 'component' : 'thumbComponent']);
        this.viewContainerRef.clear();
        var /** @type {?} */ componentRef = this.viewContainerRef.createComponent(componentFactory);
        var /** @type {?} */ galleryItem = componentRef.instance;
        galleryItem.data = this.item.data;
    };
    GalleryItemDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[item]'
                },] },
    ];
    /** @nocollapse */
    GalleryItemDirective.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
        { type: ViewContainerRef, },
    ]; };
    GalleryItemDirective.propDecorators = {
        "item": [{ type: Input },],
        "type": [{ type: Input },],
    };
    return GalleryItemDirective;
}());
export { GalleryItemDirective };
function GalleryItemDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    GalleryItemDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    GalleryItemDirective.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    GalleryItemDirective.propDecorators;
    /** @type {?} */
    GalleryItemDirective.prototype.item;
    /** @type {?} */
    GalleryItemDirective.prototype.type;
    /** @type {?} */
    GalleryItemDirective.prototype.componentFactoryResolver;
    /** @type {?} */
    GalleryItemDirective.prototype.viewContainerRef;
}
