// @ngx-gallery/core/esm5/directives/lazy.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, ElementRef, Input, Output, EventEmitter, Renderer2 } from '@angular/core';
import { Subject } from 'rxjs/Subject';
import { switchMap } from 'rxjs/operators/switchMap';
import { tap } from 'rxjs/operators/tap';
import { zip } from 'rxjs/observable/zip';
import { fromEvent } from 'rxjs/observable/fromEvent';
var LazyDirective = /** @class */ (function () {
    function LazyDirective(_el, _renderer) {
        var _this = this;
        this._el = _el;
        this._renderer = _renderer;
        this._worker$ = new Subject();
        this.loading = new EventEmitter();
        var /** @type {?} */ img = new Image();
        this._worker$.pipe(switchMap(function (imageSrc) {
            // Image is loading
            // Image is loading
            _this.loading.emit(true);
            // Stop previously loading
            img.src = imageSrc;
            // Image load success
            var /** @type {?} */ loadSuccess = fromEvent(img, 'load').pipe(tap(function () {
                _this._renderer.setStyle(_this._el.nativeElement, 'backgroundImage', "url(" + imageSrc + ")");
                _this.loading.emit(false);
            }));
            // Image load error
            var /** @type {?} */ loadError = fromEvent(img, 'error').pipe(tap(function () { return _this.loading.emit(false); }));
            return zip(loadSuccess, loadError);
        })).subscribe();
    }
    Object.defineProperty(LazyDirective.prototype, "lazyImage", {
        set: /**
         * @param {?} imagePath
         * @return {?}
         */
        function (imagePath) {
            this.loadImage(imagePath);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} imagePath
     * @return {?}
     */
    LazyDirective.prototype.loadImage = /**
     * @param {?} imagePath
     * @return {?}
     */
    function (imagePath) {
        this._worker$.next(imagePath);
    };
    /**
     * @return {?}
     */
    LazyDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this._worker$.complete();
    };
    LazyDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[lazyImage]'
                },] },
    ];
    /** @nocollapse */
    LazyDirective.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: Renderer2, },
    ]; };
    LazyDirective.propDecorators = {
        "lazyImage": [{ type: Input, args: ['lazyImage',] },],
        "loading": [{ type: Output },],
    };
    return LazyDirective;
}());
export { LazyDirective };
function LazyDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    LazyDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    LazyDirective.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    LazyDirective.propDecorators;
    /** @type {?} */
    LazyDirective.prototype._worker$;
    /** @type {?} */
    LazyDirective.prototype.loading;
    /** @type {?} */
    LazyDirective.prototype._el;
    /** @type {?} */
    LazyDirective.prototype._renderer;
}
