// @ngx-gallery/core/esm2015/directives/gallery-item.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, Input, ComponentFactoryResolver, ViewContainerRef } from '@angular/core';
export class GalleryItemDirective {
    /**
     * @param {?} componentFactoryResolver
     * @param {?} viewContainerRef
     */
    constructor(componentFactoryResolver, viewContainerRef) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.viewContainerRef = viewContainerRef;
        this.type = 'slide';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        const /** @type {?} */ componentFactory = this.componentFactoryResolver
            .resolveComponentFactory(this.item[this.type === 'slide' ? 'component' : 'thumbComponent']);
        this.viewContainerRef.clear();
        const /** @type {?} */ componentRef = this.viewContainerRef.createComponent(componentFactory);
        const /** @type {?} */ galleryItem = componentRef.instance;
        galleryItem.data = this.item.data;
    }
}
GalleryItemDirective.decorators = [
    { type: Directive, args: [{
                selector: '[item]'
            },] },
];
/** @nocollapse */
GalleryItemDirective.ctorParameters = () => [
    { type: ComponentFactoryResolver, },
    { type: ViewContainerRef, },
];
GalleryItemDirective.propDecorators = {
    "item": [{ type: Input },],
    "type": [{ type: Input },],
};
function GalleryItemDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    GalleryItemDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    GalleryItemDirective.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    GalleryItemDirective.propDecorators;
    /** @type {?} */
    GalleryItemDirective.prototype.item;
    /** @type {?} */
    GalleryItemDirective.prototype.type;
    /** @type {?} */
    GalleryItemDirective.prototype.componentFactoryResolver;
    /** @type {?} */
    GalleryItemDirective.prototype.viewContainerRef;
}
