// @thisissoon/angular-masonry/bundles/thisissoon-angular-masonry.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('@thisissoon/angular-masonry', ['exports', '@angular/core'], factory) :
    (factory((global.thisissoon = global.thisissoon || {}, global.thisissoon['angular-masonry'] = {}),global.ng.core));
}(this, (function (exports,core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    /**
     * An implemenation of the masonry library to be used
     * as a dummy when running on a server
     * @param {?} element
     * @param {?} options
     * @return {?}
     */
    function MasonryRef(element, options) {
        return {
            layout: function () { return null; },
            destroy: function () { return null; },
            stamp: function () { return null; },
            unstamp: function () { return null; },
            appended: function () { return null; },
            prepended: function () { return null; },
            addItems: function () { return null; },
            remove: function () { return null; },
            on: function () { return null; },
            off: function () { return null; },
            once: function () { return null; },
            reloadItems: function () { return null; },
            getItemElements: function () { return null; }
        };
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var /** @type {?} */ Masonry = new core.InjectionToken('Masonry');

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var ɵ0 = MasonryRef;
    var /** @type {?} */ defaultProviders = [{ provide: Masonry, useValue: ɵ0 }];
    /**
     * A simple lightweight module to use Masonry layout in Angular
     *
     * {\@link https://masonry.desandro.com/}
     *
     */
    var MasonryModule = (function () {
        function MasonryModule() {
        }
        /**
         * Specify a static method for root module to ensure providers
         * are only provided once but allows the module to still be imported
         * into other modules without reproviding services.
         *
         * \@memberof MasonryModule
         * @param {?=} providers
         * @return {?}
         */
        MasonryModule.forRoot = /**
         * Specify a static method for root module to ensure providers
         * are only provided once but allows the module to still be imported
         * into other modules without reproviding services.
         *
         * \@memberof MasonryModule
         * @param {?=} providers
         * @return {?}
         */
            function (providers) {
                if (providers === void 0) {
                    providers = defaultProviders;
                }
                return {
                    ngModule: MasonryModule,
                    providers: providers
                };
            };
        MasonryModule.decorators = [
            { type: core.NgModule },
        ];
        return MasonryModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.MasonryRef = MasonryRef;
    exports.Masonry = Masonry;
    exports.MasonryModule = MasonryModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
