// @thisissoon/angular-masonry/fesm2015/thisissoon-angular-masonry.js
import { InjectionToken, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * An implemenation of the masonry library to be used
 * as a dummy when running on a server
 * @param {?} element
 * @param {?} options
 * @return {?}
 */
function MasonryRef(element, options) {
    return {
        layout: () => null,
        destroy: () => null,
        stamp: () => null,
        unstamp: () => null,
        appended: () => null,
        prepended: () => null,
        addItems: () => null,
        remove: () => null,
        on: () => null,
        off: () => null,
        once: () => null,
        reloadItems: () => null,
        getItemElements: () => null
    };
}

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
const /** @type {?} */ Masonry = new InjectionToken('Masonry');

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
const ɵ0 = MasonryRef;
const /** @type {?} */ defaultProviders = [{ provide: Masonry, useValue: ɵ0 }];
/**
 * A simple lightweight module to use Masonry layout in Angular
 *
 * {\@link https://masonry.desandro.com/}
 *
 */
class MasonryModule {
    /**
     * Specify a static method for root module to ensure providers
     * are only provided once but allows the module to still be imported
     * into other modules without reproviding services.
     *
     * \@memberof MasonryModule
     * @param {?=} providers
     * @return {?}
     */
    static forRoot(providers = defaultProviders) {
        return {
            ngModule: MasonryModule,
            providers: providers
        };
    }
}
MasonryModule.decorators = [
    { type: NgModule },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { MasonryRef, Masonry, MasonryModule };
