// @ngx-gallery/lightbox/esm2015/lightbox.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, ChangeDetectionStrategy, HostBinding, HostListener } from '@angular/core';
import { animate, state, style, transition, trigger } from '@angular/animations';
export class LightboxComponent {
    /**
     * @param {?} e
     * @return {?}
     */
    onMouseLeave(e) {
        if (e.toState === 'void') {
            this.overlayRef.dispose();
        }
    }
}
LightboxComponent.decorators = [
    { type: Component, args: [{
                selector: 'lightbox',
                changeDetection: ChangeDetectionStrategy.OnPush,
                preserveWhitespaces: false,
                animations: [
                    trigger('slideDialog', [
                        state('in', style({ transform: 'translateY(0%) scale(1)', opacity: 1 })),
                        transition(':enter', [
                            style({ transform: 'translateY(25%) scale(0.9)', opacity: 0 }),
                            animate('400ms cubic-bezier(0.25, 0.8, 0.25, 1)')
                        ]),
                        transition(':leave', [
                            animate('400ms cubic-bezier(0.25, 0.8, 0.25, 1)', style({ transform: 'translateY(25%)', opacity: 0 }))
                        ]),
                    ])
                ],
                template: `
    <gallery [id]="id" [destroyRef]="false">
      <button class="g-btn-close" (click)="overlayRef.detach()"></button>
    </gallery>
  `
            },] },
];
/** @nocollapse */
LightboxComponent.ctorParameters = () => [];
LightboxComponent.propDecorators = {
    "slideAnimation": [{ type: HostBinding, args: ['@slideDialog',] },],
    "onMouseLeave": [{ type: HostListener, args: ['@slideDialog.done', ['$event'],] },],
};
function LightboxComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    LightboxComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    LightboxComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    LightboxComponent.propDecorators;
    /** @type {?} */
    LightboxComponent.prototype.id;
    /** @type {?} */
    LightboxComponent.prototype.overlayRef;
    /** @type {?} */
    LightboxComponent.prototype.slideAnimation;
}
