// @thisissoon/angular-masonry/esm2015/app/masonry/masonry.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule } from '@angular/core';
import { MasonryRef } from './masonry-ref';
import { Masonry } from './masonry-token';
const ɵ0 = MasonryRef;
const /** @type {?} */ defaultProviders = [{ provide: Masonry, useValue: ɵ0 }];
/**
 * A simple lightweight module to use Masonry layout in Angular
 *
 * {\@link https://masonry.desandro.com/}
 *
 */
export class MasonryModule {
    /**
     * Specify a static method for root module to ensure providers
     * are only provided once but allows the module to still be imported
     * into other modules without reproviding services.
     *
     * \@memberof MasonryModule
     * @param {?=} providers
     * @return {?}
     */
    static forRoot(providers = defaultProviders) {
        return {
            ngModule: MasonryModule,
            providers: providers
        };
    }
}
MasonryModule.decorators = [
    { type: NgModule },
];
function MasonryModule_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MasonryModule.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MasonryModule.ctorParameters;
}
export { ɵ0 };
