// @ngx-gallery/gallerize/bundles/ngx-gallery-gallerize.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@ngx-gallery/core'), require('@ngx-gallery/lightbox'), require('rxjs/Subject'), require('rxjs/observable/from'), require('rxjs/observable/empty'), require('rxjs/operators/map'), require('rxjs/operators/tap'), require('rxjs/operators/switchMap'), require('rxjs/operators/debounceTime'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('@ngx-gallery/gallerize', ['exports', '@angular/core', '@angular/common', '@ngx-gallery/core', '@ngx-gallery/lightbox', 'rxjs/Subject', 'rxjs/observable/from', 'rxjs/observable/empty', 'rxjs/operators/map', 'rxjs/operators/tap', 'rxjs/operators/switchMap', 'rxjs/operators/debounceTime', 'rxjs/operators'], factory) :
    (factory((global['ngx-gallery'] = global['ngx-gallery'] || {}, global['ngx-gallery'].gallerize = {}),global.ng.core,global.ng.common,null,null,global.rxjs.Subject,global.rxjs['observable/from'],global.rxjs['observable/empty'],global.rxjs['operators/map'],global.rxjs['operators/tap'],global.rxjs['operators/switchMap'],global.rxjs['operators/debounceTime'],global.rxjs.operators));
}(this, (function (exports,core,common,core$1,lightbox,Subject,from,empty,map,tap,switchMap,debounceTime,operators) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var GallerizeDirective = (function () {
        function GallerizeDirective(_el, _renderer, _lightbox, _gallery, platform) {
            this._el = _el;
            this._renderer = _renderer;
            this._lightbox = _lightbox;
            this._gallery = _gallery;
            this.platform = platform;
            this.gallerizer$ = new Subject.Subject();
            /**
             * Add images with specific classes to the gallery
             */
            this.gallerize = 'root';
        }
        /**
         * @return {?}
         */
        GallerizeDirective.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                var _this = this;
                /**
                 * Get galleryRef
                 */
                var /** @type {?} */ galleryRef = this._gallery.ref(this.gallerize);
                /**
                 * Serialize img classes
                 */
                var /** @type {?} */ classes = (this.forClass) ? this.forClass.split(' ').map(function (className) { return '.' + className; }) : '';
                this.gallerizer$.pipe(debounceTime.debounceTime(300), switchMap.switchMap(function () {
                    /**
                     * get all img elements from content
                     */
                    var /** @type {?} */ imageElements = _this._el.nativeElement.querySelectorAll('img' + classes);
                    if (imageElements && imageElements.length) {
                        var /** @type {?} */ images_1 = [];
                        return from.from(imageElements).pipe(map.map(function (img, i) {
                            // Add click event to the image
                            // Add click event to the image
                            _this._renderer.setStyle(img, 'cursor', 'pointer');
                            _this._renderer.setProperty(img, 'onclick', function () { return _this._lightbox.open(i, _this.gallerize); });
                            return img;
                        }), tap.tap(function (img) { return images_1.push(new core$1.ImageItem(img.src, img.src)); }), operators.finalize(function () { return galleryRef.load(images_1); }));
                    }
                    else {
                        return empty.empty();
                    }
                })).subscribe();
                // Observe content changes
                if (common.isPlatformBrowser(this.platform)) {
                    this.observer = new MutationObserver(function () { return _this.gallerizer$.next(); });
                    this.observer.observe(this._el.nativeElement, { childList: true, subtree: true });
                }
            };
        /**
         * @return {?}
         */
        GallerizeDirective.prototype.ngOnDestroy = /**
         * @return {?}
         */
            function () {
                this.gallerizer$.complete();
                if (this.observer) {
                    this.observer.disconnect();
                }
            };
        GallerizeDirective.decorators = [
            { type: core.Directive, args: [{
                        selector: '[gallerize]'
                    },] },
        ];
        /** @nocollapse */
        GallerizeDirective.ctorParameters = function () {
            return [
                { type: core.ElementRef, },
                { type: core.Renderer2, },
                { type: lightbox.Lightbox, },
                { type: core$1.Gallery, },
                { type: Object, decorators: [{ type: core.Inject, args: [core.PLATFORM_ID,] },] },
            ];
        };
        GallerizeDirective.propDecorators = {
            "gallerize": [{ type: core.Input },],
            "forClass": [{ type: core.Input },],
        };
        return GallerizeDirective;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var GallerizeModule = (function () {
        function GallerizeModule() {
        }
        GallerizeModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            core$1.GalleryModule,
                            lightbox.LightboxModule
                        ],
                        declarations: [GallerizeDirective],
                        exports: [GallerizeDirective]
                    },] },
        ];
        /** @nocollapse */
        GallerizeModule.ctorParameters = function () { return []; };
        return GallerizeModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.GallerizeDirective = GallerizeDirective;
    exports.GallerizeModule = GallerizeModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
