// @ngx-gallery/core/esm5/models/index.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function GalleryState() { }
function GalleryState_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    GalleryState.prototype.action;
    /** @type {?|undefined} */
    GalleryState.prototype.items;
    /** @type {?|undefined} */
    GalleryState.prototype.currIndex;
    /** @type {?|undefined} */
    GalleryState.prototype.hasNext;
    /** @type {?|undefined} */
    GalleryState.prototype.hasPrev;
}
/**
 * @record
 */
export function GalleryItem() { }
function GalleryItem_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    GalleryItem.prototype.data;
    /** @type {?|undefined} */
    GalleryItem.prototype.component;
    /** @type {?|undefined} */
    GalleryItem.prototype.thumbComponent;
}
/** @enum {string} */
var GalleryAction = {
    INITIALIZED: 'initialized',
    ITEMS_CHANGED: 'itemsChanged',
    INDEX_CHANGED: 'indexChanged',
    OPENED: 'opened',
    CLOSED: 'closed',
};
export { GalleryAction };
/**
 * @record
 */
export function GalleryConfig() { }
function GalleryConfig_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    GalleryConfig.prototype.gestures;
    /** @type {?|undefined} */
    GalleryConfig.prototype.loop;
    /** @type {?|undefined} */
    GalleryConfig.prototype.panSensitivity;
    /** @type {?|undefined} */
    GalleryConfig.prototype.nav;
    /** @type {?|undefined} */
    GalleryConfig.prototype.dots;
    /** @type {?|undefined} */
    GalleryConfig.prototype.counter;
    /** @type {?|undefined} */
    GalleryConfig.prototype.thumb;
    /** @type {?|undefined} */
    GalleryConfig.prototype.thumbWidth;
    /** @type {?|undefined} */
    GalleryConfig.prototype.thumbHeight;
    /** @type {?|undefined} */
    GalleryConfig.prototype.thumbPosition;
    /** @type {?|undefined} */
    GalleryConfig.prototype.disableThumb;
    /** @type {?|undefined} */
    GalleryConfig.prototype.zoomOut;
    /** @type {?|undefined} */
    GalleryConfig.prototype.slidingDirection;
    /** @type {?|undefined} */
    GalleryConfig.prototype.loadingIcon;
}
/**
 * @record
 */
export function GalleryItemComponent() { }
function GalleryItemComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    GalleryItemComponent.prototype.data;
    /** @type {?|undefined} */
    GalleryItemComponent.prototype.index;
}
