// @ngx-gallery/core/esm2015/templates/iframe-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
export class IframeItemComponent {
    /**
     * @param {?} sanitizer
     */
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
    }
}
IframeItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'iframe-item',
                changeDetection: ChangeDetectionStrategy.OnPush,
                preserveWhitespaces: false,
                template: `
    <iframe frameborder="0"
            allowfullscreen
            [src]="sanitizer.bypassSecurityTrustResourceUrl(data?.src)"></iframe>
  `,
                styles: [`
    iframe {
      position: absolute;
      width: 100%;
      height: 100%;
    }
  `]
            },] },
];
/** @nocollapse */
IframeItemComponent.ctorParameters = () => [
    { type: DomSanitizer, },
];
IframeItemComponent.propDecorators = {
    "data": [{ type: Input },],
};
function IframeItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    IframeItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    IframeItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    IframeItemComponent.propDecorators;
    /** @type {?} */
    IframeItemComponent.prototype.data;
    /** @type {?} */
    IframeItemComponent.prototype.sanitizer;
}
