// @ngx-gallery/core/esm2015/templates/image-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { Gallery } from '../services/gallery.service';
export class ImageItemComponent {
    /**
     * @param {?} gallery
     * @param {?} sanitizer
     */
    constructor(gallery, sanitizer) {
        this.gallery = gallery;
        this.sanitizer = sanitizer;
    }
}
ImageItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'image-item',
                changeDetection: ChangeDetectionStrategy.OnPush,
                preserveWhitespaces: false,
                template: `
    <div [lazyImage]="data?.src" (loading)="loading = $event"></div>
    <i class="g-loading" *ngIf="loading" [innerHTML]="sanitizer.bypassSecurityTrustHtml(gallery.config.loadingIcon)"></i>
  `,
                styles: [`
    :host {
      position: relative;
      display: block;
      width: 100%;
      height: 100%;
    }
    div {
      background-position: center center;
      background-size: cover;
      width: 100%;
      height: 100%;
    }
  `],
            },] },
];
/** @nocollapse */
ImageItemComponent.ctorParameters = () => [
    { type: Gallery, },
    { type: DomSanitizer, },
];
ImageItemComponent.propDecorators = {
    "data": [{ type: Input },],
};
function ImageItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    ImageItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    ImageItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    ImageItemComponent.propDecorators;
    /** @type {?} */
    ImageItemComponent.prototype.loading;
    /** @type {?} */
    ImageItemComponent.prototype.data;
    /** @type {?} */
    ImageItemComponent.prototype.gallery;
    /** @type {?} */
    ImageItemComponent.prototype.sanitizer;
}
