// @ngx-gallery/core/esm2015/templates/thumbnail-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
export class ThumbnailItemComponent {
}
ThumbnailItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'thumbnail-item',
                changeDetection: ChangeDetectionStrategy.OnPush,
                preserveWhitespaces: false,
                template: `
    <div [lazyImage]="data.thumb"></div>
  `,
                styles: [`
    :host {
      width: 100%;
      height: 100%;
    }
    div {
      width: 100%;
      height: 100%;
      background-size: cover;
      background-position: center center;
    }
  `]
            },] },
];
/** @nocollapse */
ThumbnailItemComponent.ctorParameters = () => [];
ThumbnailItemComponent.propDecorators = {
    "data": [{ type: Input },],
};
function ThumbnailItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    ThumbnailItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    ThumbnailItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    ThumbnailItemComponent.propDecorators;
    /** @type {?} */
    ThumbnailItemComponent.prototype.data;
}
