// @ngx-gallery/core/esm2015/templates/video-item.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
export class VideoItemComponent {
    /**
     * @param {?} sanitizer
     */
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
    }
}
VideoItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'video-item',
                changeDetection: ChangeDetectionStrategy.OnPush,
                preserveWhitespaces: false,
                template: `
    <video controls
           [src]="sanitizer.bypassSecurityTrustResourceUrl(data?.src)"
           [poster]="sanitizer.bypassSecurityTrustResourceUrl(data?.poster)"></video>
  `,
                styles: [`
    video {
      position: absolute;
      width: 100%;
      height: 100%;
    }
  `]
            },] },
];
/** @nocollapse */
VideoItemComponent.ctorParameters = () => [
    { type: DomSanitizer, },
];
VideoItemComponent.propDecorators = {
    "data": [{ type: Input },],
};
function VideoItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    VideoItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    VideoItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    VideoItemComponent.propDecorators;
    /** @type {?} */
    VideoItemComponent.prototype.data;
    /** @type {?} */
    VideoItemComponent.prototype.sanitizer;
}
